src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -127,7 +127,7 @@ class InstanceOpFactoryImpl(
     matchResponse match {
       case matches: ResourceMatchResponse.Match =>
         val maybeExistingTaskId = scheduledInstance.tasksMap.values.headOption.map(_.taskId)
-        val taskId = maybeExistingTaskId.map(Task.Id.withIncarnationCount).getOrElse {
+        val taskId = maybeExistingTaskId.map(Task.Id.nextIncarnationFor).getOrElse {
```

Comment:
Please take a look at #6681. I thought we did not want to make all the changes here.